docs/Scriptmodule-Guide.md

```diff
@@ -0,0 +1,286 @@
+
+:   Put here in quotes (`"`) a short info about where to put the ROMs, or if
+    more complex user actions are needed after installation a link to the
+    RetroPie-Docs for emulator infini. Obviously the linked documentation page
+    should exist. Newlines in the displayed text can be forced with `\n`.
+
+`rp_module_section` (predefined, pick one)
+
+:   In which section will the emulator be collated? This is relevant when the
+    user runs `retropie_setup.sh`. Most of the time you will use `exp` for
+    experimental or `opt` for optional. If an scriptmodule emulator qualifies
+    for the section `main` should be checked with the RetroPie maintainers
+    first.
+
+`rp_module_flags` (predefined, pick none to many)
+
+:   Flags, as the name suggests, control under which conditions a scriptmodule
+    can be run. Conditions may apply for the different hardwares, architectures
+    and capabilities of an architecture. For example: A flag `sdl2` requires
+    that on the target SDL2 must be available. If not the scriptmodule will not
+    run. You can also negate a flag by putting a bang (`!`) before, thus
+    `!armv6` would disallow to run the scriptmodule on Raspberry Pi Zero and
+    Raspberry Pi One. See
+    [here](https://en.wikipedia.org/wiki/Raspberry_Pi#Specifications) for
+    detailed specification. Flags have an order and are a logical And
+    conjunction. Use spaces to separate the flags. There is the special flag
+    `all`, which when put negated at the beginning can be used to create a
+    strict allow list, e.g. `!all 64bit` means this scriptmodule can only be run
+    on 64-bit architectures. Grep through the existing scriptmodules to identify
+    the current set of flags. If you assume to need a new flag, check with the
+    RetroPie maintainers first.
+
+## Major Predefined Variables
+
+`$home`
+
+:   The home directory of the account which is used to run any emulator.
+    Default: `/home/pi/`
+
+`$md_build`
+
+:   The directory where the emulator is build. Default:
+    `$home/RetroPie-Setup/tmp/build/$md_id`. Specific example: `/home/pi/RetroPie-Setup/tmp/build/infini`
+
+`$md_conf_root`
+
+:   The configuration root directory. Default: `/opt/retropie/config/`. Usually
+    supplemented with the system name e.g., `mkUserDir $md_conf_root/smaky`.
+
+`$md_data`
+
+:   Location of extra files e.g., for building or installing the emulator.
+    Mainly used to apply patches to the source tree before building.Default is
+    sibling to your scriptmodule and concatenated with the `$repo_module_id` you
+    have defined in the header (see above). For the emulator infini it would be:
+    `$home/RetroPie-Setup/scriptmodules/emulators/infini/` with the scriptmodule
+    path `$home/RetroPie-Setup/scriptmodules/emulators/infini.sh`. Scriptmodules
+    for ports and libretrocores are kept below `ports/` and `libretrocores/`
+    respectively instead of `emulators/`.
+
+`$md_id`
+
+:   Same as `$repo_module_id`
```

Comment:
there is no `repo_module_id` - there is a `rp_module_id` but it's not an available variable for use in a module so I think including it with `$` is confusing.